pkg/upgrade/upgrade.go

```diff
@@ -95,8 +96,10 @@ func Zinit(socket string) func(u *Upgrader) error {
 
 // NewUpgrader creates a new upgrader instance
 func NewUpgrader(root string, opts ...UpgraderOption) (*Upgrader, error) {
+	hubClient := hub.NewHubClient(defaultHubTimeout)
 	u := &Upgrader{
 		root: root,
+		hub:  *hubClient,
```

Comment:
when doing a `deref` (like *something) this copies the value. I think it might be better to change the `hub` attribute to `hub *hub.HubClient` instead.